src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIEmbeddingGenerator.cs

```diff
@@ -160,4 +160,18 @@ private static EmbeddingGeneratorMetadata CreateMetadata(string providerName, st
 
         return openAIOptions;
     }
+
+    private sealed class OpenAIEmbeddingGeneratorMetadata : EmbeddingGeneratorMetadata
+    {
+        private readonly EmbeddingModelMetadata _modelMetadata;
+
+        public OpenAIEmbeddingGeneratorMetadata(string? providerName, Uri? providerUri, string? defaultModelId, int? dimensions)
```

Comment:
Is there an assumption here that `dimensions` corresponds to `defaultModelId` or to any model that might be used? GetModelMetadataAsync is assuming the latter.

Reply:
Renamed to `defaultModelDimensions` for clarity.